pkg/streamingpromql/operators/selectors/instant_vector_selector.go

```diff
@@ -119,6 +120,8 @@ func (v *InstantVectorSelector) NextSeries(ctx context.Context) (types.InstantVe
 			continue
 		}
 
+		v.Stats.TotalSamples++
```

Comment:
Have we benchmarked this change?

One alternate would be to do `v.Stats.TotalSamples += v.Selector.TimeRange.EndT / v.Selector.TimeRange.IntervalMilliseconds` (probably with `floor` and an offset for the start or similar)

Or is this done here to catch stats for errored queries? I'm not sure if there would be value in that.

If not, alternatively we could look at the `len` of the returned data.